packages/composite-checkout/src/components/transaction-status-handler.ts

```diff
@@ -47,11 +48,15 @@ export function useTransactionStatusHandler( redirectToUrl: ( url: string ) => v
 	);
 	const redirectInfoMessage = __( 'Redirecting to payment partner…' );
 	useEffect( () => {
-		if ( transactionStatus === 'pending' && previousTransactionStatus !== 'pending' ) {
+		if ( transactionStatus === previousTransactionStatus ) {
```

Comment:
This change and refactoring of this hook would have been easier to review if it was in its own commit, perhaps with a small explanation. Nice find, though.